### PR TITLE
tests: filesToStreams and doFilesWrite

### PR DESCRIPTION
**Pull request by hacdias**
Here are some basic tests for the functions `filesToStreams` and `doFilesWrite`. I also made a few changes so the files are already normalised when they get to `doFilesWrite`.

### CODE REVIEW

**Comment by hacdias** [comment]
@olizilla could you please take a look? I feel I'm missing something important (everything is passing). Although there are things I don't know if and how to test if we're not inside the right environments (browsers).

**Comment by hacdias** [comment]
> 2:33 PM <olizilla[m]> hacdias: simple is a good start. it' be great to get a test that shows that the dnd workaround code that is using datatransfer-files-promise is failing right now... by mocking up the input as chrome would have passed it in

**Comment by hacdias** [comment]
> > 2:33 PM <olizilla[m]> hacdias: simple is a good start. it' be great to get a test that shows that the dnd workaround code that is using datatransfer-files-promise is failing right now... by mocking up the input as chrome would have passed it in

@olizilla I can write those tests but they probably won't fail since they would only fail on Chrome and we're using JSDom and not a true browser to run jest tests.

**Comment by hacdias** [comment]
@olizilla I think this PR is finished. Could you please review it? Let's keep them small.

File: src/lib/files.js [hunk 1]
```diff
@@ -1,37 +1,20 @@
...
-    }
-
     const stream = fileReader(file)
 
-    if (file.webkitRelativePath) {
-      isDir = true
-    }
-
     streams.push({
       path: file.filepath || file.webkitRelativePath || file.name,
```
[the remark this answers is not in the document]
Comment (olizilla) [reply src/lib/files.js]: though i guess we'd just have to mock it out to in our tests, so it's probably not going to tell us much... how do we capture the assumption we have that `file.fullPath` as set by the datatransfer-promise lib will match `file.webkitRelativePath` when we check for it here?

[the remark this answers is not in the document]
Comment (hacdias) [reply src/lib/files.js]: > In the #1049 we're checking the file.fullPath property, but here we using file.filepath || file.webkitRelativePath || file.name

Yes, we don't use `fullPath` for dragged resources, we're just using `name`.

> how do we capture the assumption we have that file.fullPath as set by the datatransfer-promise lib will match file.webkitRelativePath when we check for it here?

I'm a bit confused about what do you mean here. Could you clarify pls? 😅

[the remark this answers is not in the document]
Comment (olizilla) [reply src/lib/files.js]: When we drop a file, the test in the other PR ensures that we end up with an array of objects where each one has a `fullPath` property contains the file path for the entry 

https://github.com/ipfs-shipyard/ipfs-webui/pull/1049/files#diff-67bc186cb8bd027fa658bd5df67f0aebR118

The test in this PR checks that filesToStreams will correctly map a list of objects wiht a `webkitRelativePath` property will get mapped to a list of objects with a `path` property. So the output of the test in the other PR is not what the input to this test expects.

**Pull request merged**